src/main/java/com/minecolonies/core/entity/ai/workers/guard/KnightCombatAI.java

```diff
@@ -256,13 +257,13 @@ private double getAttackDamage()
         {
             if (heldItem.getItem() instanceof SwordItem)
             {
-                addDmg += ((SwordItem) heldItem.getItem()).getDamage(heldItem) + BASE_PHYSICAL_DAMAGE;
+                addDmg += user.getAttribute(Attributes.ATTACK_DAMAGE).getValue();
```

Comment:
It should probably do this unconditionally (i.e. remove the SwordItem/Tinkers check. since Tinkers swords should support this query too).